src/lib.rs

```diff
@@ -407,6 +380,39 @@ pub trait Rng: RngCore {
     fn sample<T, D: Distribution<T>>(&mut self, distr: D) -> T {
         distr.sample(self)
     }
+
+    /// Create an iterator that generates values using the given distribution.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use rand::{thread_rng, Rng};
+    /// use rand::distributions::{Alphanumeric, Range, Uniform};
+    ///
+    /// let mut rng = thread_rng();
+    ///
+    /// // Vec of 16 x f32:
+    /// let v: Vec<f32> = thread_rng().sample_iter(&Uniform).take(16).collect();
```

Comment:
Nice, but lets not promote bad styles in examples by creating a local `rng` handle then not using it.

Reply:
I wanted to show that it was also possible to use `thread_rng()` directly. But yes, bad style. Will change.

Reply:
I think most people will figure that out.